rust/hyperlane-base/src/types/checkpoint_syncer.rs

```diff
@@ -27,6 +27,29 @@ pub enum CheckpointSyncerConf {
 }
 
 impl CheckpointSyncerConf {
+    /// Create a CheckpointSyncerConf from a storage location string
+    pub fn from_storage_location(storage_location: &str) -> Option<Self> {
```

Comment:
I think you and I had a conversation on a different PR, where we asked the question:

"If the relayer reads invalid values written by untrusted actors, how should it respond?"

I believe that conversation was wrt validator checkpoints.

My understanding was that we settled on returning options, not errors, in these cases, is that right?

Reply:
Yeah I don't feel strongly here at all, I was thinking this is a little different because this is creating a new `CheckpointSyncerConf`, and without considering how this struct / function is intended to be used, if the `storage_location` is malformed and a `CheckpointSyncerConf` cannot be created, this feels like an error and should ideally be surfaced in some way. Otherwise I could see malformed storage locations really easily going unnoticed.

The other situations where we use `Option` I think make sense when trying to e.g. fetch data that may or may not be there, which is totally valid behavior (like when fetching signed checkpoints that haven't hit a quorum yet or something)

Reply:
Gonna leave for now we can always adjust easily later